controllers/object_controls.go

```diff
@@ -1238,15 +1238,43 @@ func TransformToolkit(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n
 
 	// configure runtime
 	runtime := n.runtime.String()
-	setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), "RUNTIME", runtime)
+	err = transformForRuntime(obj, config, runtime, "nvidia-container-toolkit-ctr")
+	if err != nil {
+		return fmt.Errorf("error transforming toolkit daemonset : %v", err)
+	}
+
+	// Update CRI-O hooks path to use default path for non OCP cases
+	if n.openshift == "" && n.runtime == gpuv1.CRIO {
+		for index, volume := range obj.Spec.Template.Spec.Volumes {
+			if volume.Name == "crio-hooks" {
+				obj.Spec.Template.Spec.Volumes[index].HostPath.Path = "/usr/share/containers/oci/hooks.d"
+			}
+		}
+	}
+	return nil
+}
+
+func transformForRuntime(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, runtime string, mainContainerName string) error {
```

Comment:
containerName should suffice as far this function is concerned
```suggestion
func transformForRuntime(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, runtime string, containerName string) error {
```

Reply:
Fixed